lms/djangoapps/canvas_integration/api.py

```diff
@@ -61,16 +90,81 @@ def sync_canvas_enrollments(course_key, canvas_course_id, unenroll_current):
     Args:
         course_key (str): The edX course key
         canvas_course_id (int): The canvas course id
-        unenroll_current (bool): If true, unenroll existing students if not staff TODO
+        unenroll_current (bool): If true, unenroll existing students if not staff
     """
-    emails = list_canvas_enrollments(canvas_course_id)
+    client = CanvasClient(canvas_course_id)
+    emails_to_enroll = client.list_canvas_enrollments()
+    users_to_unenroll = []
 
     course_key = CourseLocator.from_string(course_key)
     course = get_course_by_id(course_key)
 
-    enrolled = enroll_emails_in_course(emails=emails, course_key=course_key)
+    if unenroll_current:
+        enrolled_user_dict = {user.email: user for user in get_enrolled_non_staff_users(course)}
+        emails_to_enroll_set = set(emails_to_enroll)
+        already_enrolled_email_set = set(enrolled_user_dict.keys())
+        emails_to_enroll = emails_to_enroll_set - already_enrolled_email_set
+        users_to_unenroll = [enrolled_user_dict[email] for email in (already_enrolled_email_set - emails_to_enroll)]
+
+    enrolled = enroll_emails_in_course(emails=emails_to_enroll, course_key=course_key)
     log.info("Enrolled users in course %s: %s", course_key, enrolled)
 
-    if unenroll_current:
-        unenrolled = unenroll_non_staff_users_in_course(course)
-        log.info("Unenrolled non-staff users in course %s: %s", course_key, unenrolled)
+    if users_to_unenroll:
+        for user_to_unenroll in users_to_unenroll:
+            CourseEnrollment.unenroll(user_to_unenroll, course.id)
+        log.info("Unenrolled non-staff users in course %s: %s", course_key, users_to_unenroll)
+
+
+def push_edx_grades_to_canvas(course, canvas_course_id=None):
```

Comment:
It looks like `canvas_course_id` is never used. Can we remove it for simplicity's sake?